pybliometrics/utils/get_content.py

```diff
@@ -99,19 +108,36 @@ def get_content(url, api, params=None, **kwds):
         except (IndexError, ValueError):
             pass
 
-    # Perform request, eventually replacing the current key
-    timeout = config.getint("Requests", "Timeout", fallback=20)
-    resp = session.get(url, headers=header, proxies=proxies, params=params,
-                       timeout=timeout)
+    # Either use token header or key header
+    if token_key:
+        header = token_header
+        resp = session.get(url, headers=header, params=params, timeout=timeout)
+    else:
+        header = key_header
+        resp = session.get(url, headers=header, params=params, timeout=timeout, proxies=proxies)
+
+    # If 429 try other tokens
+    while resp.status_code == 429:
+        try:
+            insttokens.pop(0)
+            shuffle(insttokens)
+            key, token = insttokens[0]
```

Comment:
My suggestion would be to create a list of tuples combining keys and insttokens (as many keys are needed, assuming that the first key belongs to the first insttoken). Then you simply shuffle that list.